pkg/xds/topology/dataplanes.go

```diff
@@ -3,21 +3,39 @@ package topology
 import (
 	"context"
 
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	"github.com/kumahq/kuma/pkg/core/resources/manager"
 )
 
-// GetDataplanes returs list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
+// GetDataplanes returns list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
 func GetDataplanes(ctx context.Context, rm manager.ReadOnlyResourceManager, mesh string) (*core_mesh.DataplaneResourceList, error) {
 	dataplanes := &core_mesh.DataplaneResourceList{}
 	if err := rm.List(ctx, dataplanes); err != nil {
 		return nil, err
 	}
 	rv := &core_mesh.DataplaneResourceList{}
 	for _, d := range dataplanes.Items {
+		if err := ResolveAddress(d); err != nil {
+			return nil, err
+		}
 		if d.GetMeta().GetMesh() == mesh || d.Spec.IsIngress() {
 			_ = rv.AddItem(d)
 		}
 	}
 	return rv, nil
 }
+
+func ResolveAddress(dataplane *core_mesh.DataplaneResource) error {
```

Comment:
Isn't this a strange place to put this function? Anything more generic?

Reply:
Probably yes, but it depends on `dataplane` and used only inside xds package. Do you have any suggestions about another place?

Reply:
I like this one. I don't like having half of the project in `utils` directory, because it _may_ be used somewhere

Reply:
What if it is a `core_mesh.DataplaneResource` method?

Reply:
and then you are introducing DNS dependency into `core_mesh` package. I think `core` packages should not depend on other packages next to core (aside of a couple of exceptions like `api` and `utils`), but as long as Ilya will move it to a separate package so we don't drag the whole DNS server into `core_mesh` I can live with it for now ;)